examples/model_selection/randomized_search.py

```diff
@@ -40,15 +40,16 @@
 
 
 # Utility function to report best scores
-def report(grid_scores, n_top=3):
-    top_scores = sorted(grid_scores, key=itemgetter(1), reverse=True)[:n_top]
-    for i, score in enumerate(top_scores):
-        print("Model with rank: {0}".format(i + 1))
-        print("Mean validation score: {0:.3f} (std: {1:.3f})".format(
-              score.mean_validation_score,
-              np.std(score.cv_validation_scores)))
-        print("Parameters: {0}".format(score.parameters))
-        print("")
+def report(results, candidate_params, n_top=3):
+    for i in range(n_top + 1):
+        candidates = np.flatnonzero(results['default_test_rank'] == i)
```

Comment:
> I don't much like this "default". Is it worth annotating ClassifierMixin and RegressorMixin with the name of their metric?

It would be nice to do that. But a bit hacky. I'll push a commit.

> but can see benefits both ways.

What do you surmise as the benefit of `accuracy_test_rank` over `test_accuracy_rank`? I do agree that `test_accuracy_rank` or `test_split_0_accuracy` is better...